studyzen/Lectures/LectureService.cs

```diff
@@ -0,0 +1,72 @@
+using Studyzen.Lectures;
+using StudyZen.Lectures.Requests;
+using StudyZen.Persistence;
+using StudyZen.Utils;
+
+namespace StudyZen.Lectures;
+
+public interface ILectureService
+{
+    Lecture AddLecture(CreateLectureRequest request);
+    Lecture? GetLectureById(int lectureId);
+    IEnumerable<Lecture> GetLecturesByCourseId(int? courseId);
+    Lecture? UpdateLectureById(int lectureId, string? name, string? content);
+    public void DeleteLectureById(int lectureId);
+}
+
+public sealed class LectureService : ILectureService
+{
+    private readonly IUnitOfWork _unitOfWork;
+
+    public LectureService(IUnitOfWork unitOfWork)
+    {
+        _unitOfWork = unitOfWork;
+    }
+
+    public Lecture AddLecture(CreateLectureRequest request)
+    {
+        Lecture newLecture = new Lecture(request.CourseId, request.Name, request.Content);
+        _unitOfWork.Lectures.Add(newLecture);
+        return newLecture;
+    }
+    public Lecture? GetLectureById(int lectureId)
+    {
+        Lecture? requestedLecture = _unitOfWork.Lectures.GetById(lectureId);
+        return requestedLecture;
+    }
+    public IEnumerable<Lecture> GetLecturesByCourseId(int? courseId)
```

Comment:
Suggestion: return IReadOnlyCollection<Lecture>, in the LINQ add .ToList(). This way we can easily access lectures and not just enumerate through them

Reply:
suggestion applied